good-enough-practices-for-scientific-computing.tex

```diff
@@ -323,24 +332,30 @@ \subsection*{Acknowledgments}
 
 \item
   \practice{1d}{Record all the steps used to process data}: Data manipulation is as integral
-  to your analysis as statistical modelling and inference. If you do not document this
+  to your analysis as statistical modeling and inference. If you do not document this
   step thoroughly, it is impossible for you, or anyone else, to repeat the analysis.
 
   The best way to do this is to write scripts for \emph{every} stage of data processing.
   This might feel frustratingly slow, but you will get faster with practice. The
   immediate payoff will be the ease with which you can re-do data preparation when new
   data arrives. You can also re-use data preparation steps in the future for related
-  projects. For very large data sets, data preparation may also include scripts to obtain 
+  projects. For very large data sets, data preparation may also include writing and saving scripts to obtain 
   the data or subsets of the data from remote storage. 
 
-  When scripting is not feasible, it's important to clearly document every manual action
+  Some data cleaning tools, such as \withurl{OpenRefine}{http://www.openrefine.org}, 
+  % check OpenRefine URL
+  provide a graphical user interface, but
+  also automatically keep track of each step in the process. When tools like these or
+  scripting is not feasible, it's important to clearly document every manual action
   (what menu was used, what column was copied and pasted, what link was clicked, etc.).
   Often you can at least capture \emph{what} action was taken, if not the complete
   \emph{why}. For example, choosing a region of interest in an image is inherently
   interactive, but you can save the region chosen as a set of boundary coordinates.
+  
+ 
 
 \item
-  \practice{1e}{Anticipate the need to use multiple tables}: Raw data, even if tidy, is not
+  \practice{1e}{Anticipate the need to use multiple tables, and use unique identifiers}: Raw data, even if tidy, is not
```

Comment:
"and use a unique identifier for each record" - otherwise, "use unique identifiers" can be read to mean "unique identifier for each table"